### PR TITLE
#1656 Allows 0 expanded facets

### LINKED ISSUE #1656: Number of expanded facets is always > 0
Acoording to settings, there is a possibility to set number of facets that are initially opened. In case you want all of them kept closed (set 0 in administration), this setting is not respected.

### Preconditions

Magento Version : 2.3.1

ElasticSuite Version : 2.8.0

Environment : doesn't matter

Third party modules :

### Steps to reproduce

1. Set number of initially opened facets to 0
2. Clear caches

### Expected result

1. None of the present facets should be opened 

### Actual result

1. At least one (the first) is opened

### PR DESCRIPTION
**Pull request by NORDCOMPCZ**
Fixes #1656 Allows 0 expanded facets
See https://github.com/Smile-SA/elasticsuite/issues/1656 for more information

### CODE REVIEW

**Comment by rbayet** [comment]
Hello @NORDCOMPCZ,

Could you amend your commit message to include a brief description of the fix ?
For instance :
"Fixes #1656 Allows 0 expanded facets"

Regards,

**Comment by NORDCOMPCZ** [comment]
@rbayet updated (commit & PR msg)

**Comment by rbayet** [comment]
Hello @NORDCOMPCZ,

Your fix works as expected, could you fix the two issues phpcs Travis identified ?
```
FILE: ...Smile-SA/elasticsuite/src/module-elasticsuite-catalog/Block/Navigation.php
--------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 2 LINES
--------------------------------------------------------------------------------
 120 | WARNING | Please trim any trailing whitespace
 124 | WARNING | Please trim any trailing whitespace
--------------------------------------------------------------------------------
```

Regards,

File: src/module-elasticsuite-catalog/Block/Navigation.php [hunk 1]
```diff
@@ -116,8 +116,12 @@ public function getActiveFilters()
             $requestParams    = array_keys($this->getRequest()->getParams());
             $displayedFilters = $this->getDisplayedFilters();
             $expandedFacets   = $this->_scopeConfig->getValue(self::DEFAULT_EXPANDED_FACETS_COUNT_CONFIG_XML_PATH);
-            $activeFilters    = range(0, min(count($displayedFilters), $expandedFacets) - 1);
-
+            $activeFilters = [];
```
Comment (rbayet) [inline src/module-elasticsuite-catalog/Block/Navigation.php]: Could you align the = operator with those above ?

File: src/module-elasticsuite-catalog/Block/Navigation.php [hunk 2]
```diff
@@ -116,8 +116,12 @@ public function getActiveFilters()
...
+
```
Comment (rbayet) [inline src/module-elasticsuite-catalog/Block/Navigation.php]: That line can be removed.

File: src/module-elasticsuite-catalog/Block/Navigation.php [hunk 3]
```diff
@@ -116,8 +116,12 @@ public function getActiveFilters()
...
             $displayedFilters = $this->getDisplayedFilters();
             $expandedFacets   = $this->_scopeConfig->getValue(self::DEFAULT_EXPANDED_FACETS_COUNT_CONFIG_XML_PATH);
-            $activeFilters    = range(0, min(count($displayedFilters), $expandedFacets) - 1);
-
+            $activeFilters = [];
+            
+            if ($expandedFacets > 0) {
+                $activeFilters = range(0, min(count($displayedFilters), $expandedFacets) - 1);
+            }
+
```
Comment (rbayet) [inline src/module-elasticsuite-catalog/Block/Navigation.php]: Please remove any trailing spaces.

File: src/module-elasticsuite-catalog/Block/Navigation.php (hunk 2 above)
Comment (rbayet) [inline src/module-elasticsuite-catalog/Block/Navigation.php]: Could your align the = operator and remove the following empty line ?

File: src/module-elasticsuite-catalog/Block/Navigation.php (hunk 3 above)
Comment (rbayet) [inline src/module-elasticsuite-catalog/Block/Navigation.php]: Please remove any trailing space on that empty line.

**Pull request merged**